docs/source/kinds_of_types.rst

```diff
@@ -356,6 +356,43 @@ check against ``None`` in the if condition.
 
     ``--strict-optional`` is experimental and still has known issues.
 
+.. _noreturn:
+
+The NoReturn type
+*****************
+
+Mypy also provides an experimental support for functions that supposed to
+never return, such as ``sys.exit()``. For example, a function that raises
+an exception:
+
+.. code-block:: python
+
+   from mypy_extensions import NoReturn
+
+   def stop() -> NoReturn:
+       raise Exception('no way')
+
+Mypy will recognize that the code after such function is unreachable and will
+behave accordingly:
+
+.. code-block:: python
+
+   def f(x: int) -> int:
+       if x == 0:
+           return x
+       stop()
+       return 'whatever works'  # No error in an unreachable block
+
+Note that ``NoReturn`` annotation is different from just not returning
```

Comment:
This is a run-on sentence and "Note that" doesn't add anything. Maybe rewrite this as "Annotating a function with ``NoReturn`` is different from just not returning a value from a function. In the latter case, ``None`` is returned implicitly."